### ISSUE TITLE
Cannot navigate dropdown when NVDA is active

### ISSUE THREAD

**Issue by Nazerbayev** [paypal/downshift#458]
Using keyboard navigation only on firefox 60.0.2 (not verified with other browsers) after opening the dropdown menu or autocomplete options are displayed, if trying to navigate between the options using the arrow keys the menu is closed after de-selecting the first item on the options

* `downshift` version: 1.31.15

**Comment by kentcdodds** [comment]
Thank you for finding this issue! I don't have a Windows machine or NVDA so we'll need you help to track down the cause is this issue and resolve it. Could you please?

**Comment by Nazerbayev** [comment]
i'll look it up on my spare time

**Comment by Nazerbayev** [comment]
i've an update @kentcdodds : the issue seems to be related with how NVDA traverses through the pages. If the currently focused element isnt an input (link, button, etc) then using the arrow keys allows you to change the focus as if the tab key was used, if the element is an input then arrow keys are not overriden.

So, the issue might not be with downshift per se but how we render the dropdown (since render is delegated to the dev). Feel free to close this issue in the meantime until I do further testing, if a find a solution within the code i will reopen this ticket. If the issue is with how we render the dropdown then I will make a pull request with a note in the documentation/readme with a warning/solution.

**Comment by kentcdodds** [comment]
Thank you so much for doing this work @Nazerbayev! This will help downshift and everyone using it be more accessible. Please do let me know what you find out and we'll try to encode a solution into downshift. I'm guessing it will require that we add a prop to the input or something. If that's the case, then it's an easy fix and that would go here:

https://github.com/paypal/downshift/blob/1978146478136b29381350cf7245c96e514f1c9b/src/downshift.js#L685-L698

Can't wait to hear more of what you can uncover. Thank you for what you can do here :)

**Comment by Nazerbayev** [comment]
actually i think we need to remove `aria-activedescendant`, according to mdn:
```
Possible effects on user agents and assistive technology 

The user agent, which is any software that retrieves, renders and facilitates end user interaction with Web content, uses the aria-activedescendant property to inform the assistive technology about the active child which has focus. This active child using the aria-activedescendant property must always be visible on screen and must be a descendant of the container in the Document Object Model.
```
I think the main issue is that the options are not direct descendant of the input that receives the active-descendant prop. After i removed that prop on my sample code nvda did swap to forms mode as intented (nvda wasnt changing modes even if the input had role=combobox)

This still needs further testing though, right now i only have access to a windows machine but by next week i can have access to an osx device and i can test if removing that prop wont break voiceover

**Comment by kentcdodds** [comment]
Huh, I wonder if we could put the `aria-activedescendant` on the root div instead if that would fix the issue with NVDA without causing issues for other screen readers 🤔

**Comment by alexandernanberg** [comment]
I've done some digging and it seems like NVDA doesn't fire `keyDown` event on buttons when you are in "browse mode". There is an issue about this on their GitHub page https://github.com/nvaccess/nvda/issues/7898. 

There is a workaround and that is to enter "focus mode", typically by pressing `caps`+`space` then the `keyDown` events will fire and Downshift works as expected. 

I'm uncertain if there is anything we can do to solve this?

**Comment by Nazerbayev** [comment]
About nvda swallowing key events it is normal that it does that on browse mode, nvda was failing to change to forms mode when reading the role of combobox on <2.0.x

2.0.x+ is using listbox and it seems to be working as expected

**Comment by kentcdodds** [comment]
Hey that's great! Thanks for working through this together team.

So if I'm not missing something, 2.0.0 fixed the issue and we're good to go now right? If so, please close this issue :)
